lms/djangoapps/verify_student/tests/test_views.py

```diff
@@ -885,12 +895,14 @@ def test_create_order_already_verified(self):
 
         # Verify that the information will be sent to the correct callback URL
         # (configured by test settings)
-        data = json.loads(response.content)
-        self.assertEqual(data['override_custom_receipt_page'], "http://testserver/shoppingcart/postpay_callback/")
-
-        # Verify that the course ID and transaction type are included in "merchant-defined data"
-        self.assertEqual(data['merchant_defined_data1'], unicode(self.course.id))
-        self.assertEqual(data['merchant_defined_data2'], "verified")
+        self._check_payment_data(
+            response,
+            {
+                'override_custom_receipt_page': "http://testserver/shoppingcart/postpay_callback/",
+                'merchant_defined_data1': unicode(self.course.id),
+                'merchant_defined_data2': "verified",
```

Comment:
Good point.  I ended up substantially rewriting this test so that it would run the same requests against both backends.  I did remove everything that differentiates between verified and non-verified users.  The code/tests in this package need even more cleanup work but that's best saved for subsequent PRs.